### PR TITLE
Staging PR voting/deleting questions

### PR DESCRIPTION
**Pull request by SpencerCDixon**

### CODE REVIEW

File: app/models/question.rb [hunk 1]
```diff
@@ -44,6 +44,10 @@ def self.search(query)
     where("searchable @@ plainto_tsquery(?)", query)
   end
 
+  def destroyable_by?(user)
```
Comment (dpickett) [inline app/models/question.rb]: this can cause a `NoMethodError` on nil when the user is signed out. Write a passing unit test that considers the `nil` boundary condition and this can be merged in

File: app/views/questions/show.html.haml [hunk 2]
```diff
@@ -43,7 +43,7 @@
         = link_to @question.user.username, user_path(@question.user)
         asked on
         .created_at= format_datetime(@question.created_at)
-        - if @question.user == current_user
+        - if @question.user == current_user || current_user.try(:admin?)
```
Comment (dpickett) [inline app/views/questions/show.html.haml]: this should utilize the `destroyable_by?` and `editable_by?` instance methods on the `Question`

File: spec/models/question_spec.rb [hunk 3]
```diff
@@ -93,4 +93,29 @@
...
+
+    context "admin" do
+      it 'returns true' do
+        expect(question.destroyable_by?(ee)).to be true
+      end
+    end
+
+    context "student who wrote question" do
+      it 'returns true' do
+        expect(question.destroyable_by?(author)).to be true
```
Comment (dpickett) [inline spec/models/question_spec.rb]: rspec has some neat syntactic sugar for you.

This is functionally equivalent and reads better:

```ruby
expect(question).to be_destroyable_by(author)
```
